packages/modals/src/elements/DrawerModal/DrawerModal.tsx

```diff
@@ -112,19 +112,20 @@ export const DrawerModal = forwardRef<
         return undefined;
       }
 
-      const bodyElement = environment.querySelector('body');
+      const htmlElement = environment.querySelector('html');
+      let previousHtmlOverflow: string;
 
-      if (bodyElement && isOpen) {
-        const previousBodyOverflow = bodyElement.style.overflow;
+      if (htmlElement && isOpen) {
+        previousHtmlOverflow = htmlElement.style.overflow;
 
-        bodyElement.style.overflow = 'hidden';
-
-        return () => {
-          bodyElement.style.overflow = previousBodyOverflow;
-        };
+        htmlElement.style.overflow = 'hidden';
       }
 
-      return undefined;
+      return () => {
+        if (htmlElement) {
+          htmlElement.style.overflow = previousHtmlOverflow;
```

Comment:
Could this unintentionally remove inline overflow styling on the `html`? The `return undefined` was intentional I think

Reply:
> The return undefined was intentional I think

I figured it was intentionally added because the project has an ESLint rule that disallows implicit returns on arrow functions. 

> Could this unintentionally remove inline overflow styling on the html?

I don't think so, do you see a scenario where that might happen?

Reply:
Since `previousHtmlOverflow` is undefined when the DrawerModal is closed:
1. Include `DrawerModal` in your page
2. Unmount the `DrawerModal` without applying the `isOpen` prop
3. The `useEffect` return statement is called during unmount. Since `htmlElement` is defined it will apply `htmlElement.style.overflow = undefined;`

Wouldn't this wipe out any inline overflow styling?

Reply:
The purpose of the return is to undo the `hidden` styling applied when `isOpen=true`. Resetting it when `isOpen=false` doesn't feel right.

Reply:
> Since htmlElement is defined it will apply htmlElement.style.overflow = undefined;

I think it actually applies `previousHtmlOverflow` which is whatever the overflow styling was prior so there shouldn't be any noticeable difference from a user perspective. 

However, I added a new commit to only run the effect when the drawer modal `isOpen={true}` since that's the only time the cleanup is necessary 👍 .